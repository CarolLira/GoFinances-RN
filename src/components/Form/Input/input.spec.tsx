import React from 'react';
import { render } from '@testing-library/react-native';

import { Input } from './index';
import { ThemeProvider } from 'styled-components/native';
import theme from '../../../global/styles/theme';

const Providers: React.FC = ({ children }) => (
    <ThemeProvider theme={theme}>
        {children}
    </ThemeProvider>
);

describe('Input (component)', () => {
    it('should have specific border color when active', () => {
        const { getByTestId, toJSON } = render(
            <Input
                testID='input-email'
                placeholder='E-mail'
                keyboardType='email-address'
                autoCorrect={false}
            />,
            {
                wrapper: Providers
            }
        );
        expect(toJSON()).toMatchSnapshot();
    });
});