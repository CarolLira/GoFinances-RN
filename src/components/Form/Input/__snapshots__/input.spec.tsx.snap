// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input (component) should have specific border color when active 1`] = `
<TextInput
  allowFontScaling={true}
  autoCorrect={false}
  keyboardType="email-address"
  placeholder="E-mail"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 5,
        "color": "#000000",
        "fontFamily": "Poppins_400Regular",
        "fontSize": 27,
        "marginBottom": 8,
        "paddingBottom": 16,
        "paddingLeft": 18,
        "paddingRight": 18,
        "paddingTop": 16,
        "width": "100%",
      },
    ]
  }
  testID="input-email"
  underlineColorAndroid="transparent"
/>
`;
